utils/build-script-impl

```diff
@@ -2328,8 +2496,8 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_COMPILE_TOOLS_DEPLOYMENT_TARG
                 esac
 
                 echo "--- Installing ${product} ---"
-                XCTEST_BUILD_DIR=$(build_directory ${deployment_target} xctest)
-                XCTEST_INSTALL_PREFIX="${INSTALL_DESTDIR}/${INSTALL_PREFIX}/lib/swift/${LIB_TARGET}"
+                XCTEST_BUILD_DIR=$(build_directory ${host} xctest)
+                XCTEST_INSTALL_PREFIX="${host_install_destdir}/${INSTALL_PREFIX}/lib/swift/${LIB_TARGET}"
```

Comment:
on OSX, `host_install_destdir` will be the path to the toolchain root, and `INSTALL_PREFIX` will always be `/usr`, not the full path to the toolchain.

This should resolve to `/destdir/toolchain-prefix/usr/lib/swift/macosx`, which is correct